packages/core/src/Actor.ts

```diff
@@ -169,31 +160,36 @@ export class Actor<TEvent extends EventObject, TEmitted> {
     this.current = behavior.initial;
   }
   public start() {
-    this.behavior = this.behavior.receiveSignal(this.context, startSignal);
+    this.current = this.behavior.receive(
+      this.current,
+      startSignal,
+      this.context
+    );
     return this;
   }
   public stop() {
-    this.behavior = this.behavior.receiveSignal(this.context, stopSignal);
+    this.mailbox.length = 0; // TODO: test this behavior
+    this.current = this.behavior.receive(
+      this.current,
+      stopSignal,
+      this.context
+    );
   }
   public subscribe(observer) {
     return this.behavior.subscribe?.(observer) || nullSubscription;
   }
-  public receive(event) {
+  public receive(event: TEvent | LifecycleSignal) {
```

Comment:
similarly here - `LifecycleSignal` shouldn't be passed manually here, it should always come from start/stop methods
```suggestion
  public receive(event: TEvent) {
```